novaagent/libs/centos.py

```diff
@@ -167,13 +167,17 @@ def _os_defaults_network_manager():
         :return: has network manager only, not network scripts
         """
         dist = distro.id()
-        server_os_version = ServerOS.version_tuple()
+        server_os_version = ServerOS.version_float()
 
         log.info("Linux Distribution Detected: {0} Version {1}".format(
-            dist, ServerOS.version_tuple()))
-        if dist in ['rhel', 'centos'] and server_os_version >= (8, 0):
+            dist, ServerOS.version_float()))
+        if dist == 'rhel' and server_os_version >= 8:
             return True
-        if dist == 'fedora' and server_os_version >= (29,):
+
+        if dist == 'centos' and server_os_version >= 8:
+            return True
+
+        if dist == 'fedora' and server_os_version >= 29:
```

Comment:
This could be simplified quite a bit if we just use `int(distro.major_version())` for the comparison.  That would let us just remove the version_tuple/version_float method entirely.  It would also simplify the unit tests.